plugins/modules/zos_job_submit.py

```diff
@@ -1108,7 +1108,9 @@ def assert_valid_return_code(max_rc, job_rc, ret_code, result):
     # should NOT be 'changed=true' even though the user did override the return code,
     # a non-zero return code means the job did not change anything, so set it as
     # result["chagned"]=False,
-    if job_rc != 0:
+    if max_rc and job_rc > max_rc:
```

Comment:
Could you explain this if statement? By reading it I understand that if both `max_rc` and `job_rc` are greater that `max_rc` we return False, but by testing I see this behavior : 

```python
Python 3.10.9 (main, Mar  1 2023, 12:20:14) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> bool(0 and 12 > 8)
False
>>> bool(0 and 8 > 12)
False
>>> bool(4 and 8 > 12)
False
>>> bool(0 and 0 > 12)
False
>>> if 4 and 8 > 12:
...     print("hi")
... else:
...     print("nope")
... 
nope
>>> bool(0 and 0 > 0)
False
>>> bool(0 and 0 > -1)
False
```

I ran multiple tests but I don't see a case where it is True, because if whatever is evaluated at the `and`  section is True (or 1) we need a negative or 0 value in max_rc to be True right? that assumption might be wrong on my end so please explain

Reply:
There are some job that a return code not 0 can be success depends on the user and job, that's why the option of `max_rc`  but in these case if the value is gave (2) and the job returns a value over 2 (>2) is that not work for the user so it fails other case as the rc come 1-2 is a success your examples should be:
```
>>> bool(0 and 12 > 0)
True
>>> bool(0 and 8 > 0)
True
>>> bool(4 and 8 > 4)
True
>>> if 4 and 8 > 4:
...     print("hi")
... else:
...     print("nope")
... 
hi
```
The test `test_job_submit_max_rc` comes with a example, the job return a code of 8 if the `max_rc` is 0 or 4, the rc will be over and set to false, but when `max_rc` is 12  is fine for the user
```
bool(12 and 8 > 12):
False
```